src/common/post.ts

```diff
@@ -220,6 +225,41 @@ export const createFreeformPost = async (
   return con.getRepository(FreeformPost).save(createdPost);
 };
 
+export type CreateSourcePostModeration = Omit<
+  CreatePost,
+  'authorId' | 'content' | 'contentHtml'
+> &
+  Pick<
+    SourcePostModeration,
+    'titleHtml' | 'content' | 'type' | 'sharedPostId' | 'createdById'
+  > & {
+    contentHtml?: string;
+    externalLink?: string;
+  };
+
+export const createSourcePostModeration = async (
+  con: DataSource | EntityManager,
+  args: CreateSourcePostModeration,
+) => {
+  const newPost = con.getRepository(SourcePostModeration).create({
+    ...args,
+    status: SourcePostModerationStatus.Pending,
+  });
+  await con.getRepository(SourcePostModeration).save(newPost);
+  return newPost;
```

Comment:
We should return the result from `save` because it will have PK and other defaults from the DB.